src/plotman/archive.py

```diff
@@ -49,8 +49,14 @@ def compute_priority(phase, gb_free, n_plots):
 
 def get_archdir_freebytes(arch_cfg):
     archdir_freebytes = { }
-    df_cmd = ('ssh %s@%s df -aBK | grep " %s/"' %
-        (arch_cfg['rsyncd_user'], arch_cfg['rsyncd_host'], arch_cfg['rsyncd_path']) )
+    arch_mode = arch_cfg.get('mode', 'remote')
+    if arch_mode == 'remote':
+        df_cmd = ('ssh %s@%s df -BK | grep " %s/"' %
+            (arch_cfg['rsyncd_user'], arch_cfg['rsyncd_host'], arch_cfg['rsyncd_path']) )
+    elif arch_mode == 'local':
+        df_cmd = ('df -BK | grep " %s/"' % arch_cfg['rsyncd_path'] )
```

Comment:
There's some duplication here that makes me itchy.  But I think the piping and grepping add some complexity here.  In one case the `df` and `grep` are on different systems and on the other they are on the same system.  But I think really I'd like to see the pipe dropped entirely.  `[line for line in lines if f" {arch_cfg['rsyncd_path']}/" in line]` is a small price to pay for not needing to use a shell when subprocessing.  (or something fancier maybe if I thought about what was actually being done)  Not to mention that we had some hazards there since we weren't escaping the path for `grep`...

Reply:
I can deal with the `grep` removal separately.  No need to hold this up on that.